components/eam/src/chemistry/mozart/mo_chm_diags.F90

```diff
@@ -1537,41 +1558,55 @@ subroutine gaschmmass_diags( lchnk, ncol, vmr, vmr_old, pdeldry, mbar, rdelt, fl
                wrk(:ncol,:) = adv_mass(m)*(vmr(:ncol,:,m)-vmr_old(:ncol,:,m)) &
                                 /mbar(:ncol,:)*pdeldry(:ncol,:)*rgrav*rdelt
             endif
- 
-            if (len(flag) >= 6 .and. flag(6:8) == '_LL') then
-               wrk_sum(:ncol) = 0.0_r8
-               do k = gaschmbudget_2D_L1_s, gaschmbudget_2D_L1_e
-                  wrk_sum(:ncol) = wrk_sum(:ncol) + wrk(:ncol,k)
-               enddo
-               call outfld( trim(solsym(m))//'_'//flag(1:5)//'_L1', wrk_sum(:ncol), ncol ,lchnk )
+            endif !kzm++
+
+            !kzm ++ 
+            !if (len(flag) >= 6 .and. flag(6:8) == '_LL') then
+            if (len(flag) >= 6) then
+                    if (flag(6:8) == '_LL') then
```

Comment:
What if len(flag) < 8 (while >=6 satisfied)? The sub-string reference would still be problematic.

Reply:
@wlin7 Yes, len(flag) >=8 would be better here